src/forms/SubmitButton.tsx

```diff
@@ -0,0 +1,30 @@
+import { ObjectState } from "@homebound/form-state";
+import { Button, ButtonProps, useComputed } from "src";
+
+export type SubmitButtonProps<T> = Omit<ButtonProps, "label"> & {
+  label?: ButtonProps["label"];
+  form: ObjectState<T>;
+};
+
+/** Provides a Button that will auto-disable if `formState` is invalid. */
+export function SubmitButton<T>(props: SubmitButtonProps<T>) {
+  const { form, disabled, onClick, label = "Submit", ...others } = props;
+  const invalidAndTouched = useComputed(() => !form.valid && form.touched, [form]);
```

Comment:
**question:** Any benefit to using the `useComputed` vs wrapping the Button in an `<Observer>`?

Reply:
I think the `useComputed`s just end up being cleaner, as they don't have that ~4-lines of Observer+lambda in the jsx.